plugins/modules/web_infrastructure/jira.py

```diff
@@ -410,6 +416,7 @@ class JIRA(StateModuleHelper):
             ),
             username=dict(type='str', required=True),
             password=dict(type='str', required=True, no_log=True),
```

Comment:
Don't you want to get rid of the required username/password arguments?
```suggestion
            username=dict(type='str'),
            password=dict(type='str', no_log=True),
```
If yes, you need to make them required together, make token and password mutually exclusive, and require at least one of token and password. (Grep for `required_together`, `required_one_of` and `mutually_exclusive` in plugins/modules to find out how to use them.)

Reply:
Thank you for the suggestion! YesI think that approach makes sense and I thought that would make more sense.  I will research the proper syntax for doing that and submit a new commit which includes the suggestion.

Reply:
Added the following:

`+        mutually_exclusive=[
+            ['username', 'token'],
+            ['password', 'token'],
+        ],
+        required_together=[
+            ['username', 'password'],
+        ],
+        required_one_of=[
+            ['username', 'token'],
+        ],
`